### ISSUE TITLE
Custom CA Integration using Kubernetes CSR does not work with intermediate CAs

### ISSUE THREAD

**Issue by eoftedal** [istio/istio#33547]
**Bug description**
I have setup the cluster following https://istio.io/latest/docs/tasks/security/cert-management/custom-ca-k8s/ part2, but I am instead using a custom signer. The custom signer signs the workload certificates with an intermediate ca. I have put the root ca certificate in external-ca-cert as root-cert.pem. When the signer issues certificates it includes the workload certificate and the intermediate certficiate in status.certificate. This does not work (istiod claims the certificate is signed by an unknown authority). As a test I also put the intermediate + root ca in root-cert.pem. In this case, istiod accepted the certificate, but used both certs as root cert within the cluster, which isn't correct. I cannot find a way to specify the intermediate certificate in external-ca-cert. I tried to call it ca-cert.pem, but to no avail.
So even though istiod knows about the root CA, and gets both the workload and intermediate CA cert from status.certificate, it fails to validate the certificate chain.

[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ x ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure
[ ] Upgrade

**Expected behavior**

It should accept the signed certificate when it includes an intermediate CA. Either it should be possible to specify it in external-ca-cert or it istiod should recognize that it is a part of the status.certficiate on CSR.

**Steps to reproduce the bug**

**Version** 
istio v1.10
kubernetes 1.18.20+

**How was Istio installed?**
istiod : https://istio.io/latest/docs/tasks/security/cert-management/custom-ca-k8s/ part2

**Environment where the bug was observed (cloud vendor, OS, etc)**
minikube

**Comment by eoftedal** [comment]
After digging though the code it seems [NewKeyCertBundleWithRootCertFromFile](https://github.com/istio/istio/blob/195e98f7d6619217365c4d731ef11d834f025525/security/pkg/pki/util/keycertbundle.go#L93) only reads the root-cert and nothing else. Maybe it should also read a signing cert and chain?

**Comment by eoftedal** [comment]
It also tries to parse a single cert at https://github.com/istio/istio/blob/da6178604559bdf2c707a57f452d16bee0de90c8/security/pkg/k8s/chiron/utils.go#L340 while it should ideally parse a chain. `status.certificate` can contain a chain, not just the cert. From https://kubernetes.io/docs/reference/access-authn-authz/certificate-signing-requests/ we can read:

*Commonly, the status.certificate field contains a single PEM-encoded X.509 certificate once the CSR is approved and the certificate is issued. Some signers store multiple certificates into the status.certificate field. In that case, the documentation for the signer should specify the meaning of additional certificates; for example, this might be the certificate plus intermediates to be presented during TLS handshakes.*

**Comment by shankgan** [comment]
duplicate of https://github.com/istio/istio/issues/34218
